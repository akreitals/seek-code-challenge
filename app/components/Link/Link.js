import styled from 'styled-components';
import PropTypes from 'prop-types';
import Text from '../Text/Text';
import theme from '../theme';

const Link = styled(Text.withComponent('a'))`
    text-decoration: ${props => (props.underline ? 'underline' : 'none')};
    cursor: pointer;

    &:hover {
        text-decoration: underline;
    }
`;

Link.propTypes = {
    href: PropTypes.string,
    underline: PropTypes.bool
};

Link.defaultProps = {
    theme,
    color: 'info'
};

export default Link;
