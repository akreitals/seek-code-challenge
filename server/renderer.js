import React from 'react';
import ReactDomServer from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { configureStore } from '../app/state/configureStore';
import ThemeProvider from '../app/components/ThemeProvider/ThemeProvider';

import RoutesContainer from '../app/containers/Routes/RoutesContainer';

const isProduction = process.env.NODE_ENV === 'production' || false;

const render = (html, styles) => {
    const assetsManifest =
        process.env.webpackAssets && JSON.parse(process.env.webpackAssets);

    return `
    <!DOCTYPE html>
    <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1">
            <meta http-equiv="x-ua-compatible" content="ie=edge">
            <title>Seek Coding Challenge</title>
            ${styles}
        </head>
        <body>
            <div id="root">${html}</div>
            <script src='${
                isProduction ? assetsManifest['/vendor.js'] : 'js/vendor.js'
            }'></script>
            <script src='${
                isProduction ? assetsManifest['/app.js'] : 'js/app.js'
            }'></script>
        </body>
    </html>
    `;
};

const serverSideRender = (req, res) => {
    const context = {};
    const sheet = new ServerStyleSheet();
    const store = configureStore();

    const appWithRouter = (
        <Provider store={store}>
            <ThemeProvider>
                <StaticRouter location={req.url} context={context}>
                    <RoutesContainer />
                </StaticRouter>
            </ThemeProvider>
        </Provider>
    );

    if (context.url) {
        return res.redirect(context.url);
    }

    const html = ReactDomServer.renderToString(
        sheet.collectStyles(appWithRouter)
    );

    const styles = sheet.getStyleTags();

    return res.status(200).send(render(html, styles));
};

export default serverSideRender;
