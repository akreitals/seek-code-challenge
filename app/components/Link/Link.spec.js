import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Link from './Link';

describe('Link', () => {
    test('should render a <Link /> component', () => {
        const json = renderer.create(<Link>Hello</Link>).toJSON();
        expect(json).toMatchSnapshot();
    });

    test('should render a <Link /> component with an underline prop', () => {
        const json = renderer.create(<Link underline>Underline</Link>).toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('text-decoration', 'underline');
    });
});
