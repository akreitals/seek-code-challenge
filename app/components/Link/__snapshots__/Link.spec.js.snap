// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render a <Link /> component 1`] = `
.c1 {
  margin: 0px;
  color: #2765cf;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<a
  className="c0 c1"
  color="info"
>
  Hello
</a>
`;

exports[`Link should render a <Link /> component with an underline prop 1`] = `
.c1 {
  margin: 0px;
  color: #2765cf;
}

.c0 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<a
  className="c0 c1"
  color="info"
>
  Underline
</a>
`;
